test/distributed/test_c10d_spawn_gloo.py

```diff
@@ -35,8 +35,11 @@ def _init_pg_gloo(cls, rank, filename, world_size):
             # set process group backends manually
             c10d.init_process_group(backend="gloo", store=store, rank=rank, world_size=world_size)
             pg = c10d.distributed_c10d._get_default_group()
-            pg._set_backend(torch.device("cpu"), c10d.Backend.GLOO, backend)
-            pg._set_backend(torch.device("cuda"), c10d.Backend.GLOO, backend)
+            try:
+                pg._set_backend(torch.device("cpu"), c10d.Backend.GLOO, backend)
+                pg._set_backend(torch.device("cuda"), c10d.Backend.GLOO, backend)
+            except AttributeError:
+                print('Attribute _set_backend not found')
```

Comment:
c

Reply:
@H-Huang It seems you had added the `_set_backend` lines in your PR https://github.com/pytorch/pytorch/pull/88330. However, IIUC, those lines are only being tested on ROCm due to the python < 3.9 check, and they give an error `object has no attribute '_set_backend'`, which is not a ROCm-specific error afaict. Can you please suggest if there's a better fix that doesn't involve ROCm-specific logic?